packages/legacy-client/src/client/interfaces/filters.test.ts

```diff
@@ -20,15 +20,22 @@ import type {
   NumericFilter,
   StringFilter,
 } from "../../ontology-runtime";
-import type { Todo } from "../../util/test/TodoObject";
+import type { MockOntology } from "../../util/test";
 import type { ObjectTypeFilter } from "./filters";
 
 describe("Filters", () => {
   it("correct types", () => {
-    expectTypeOf<ObjectTypeFilter<Todo>>().toMatchTypeOf<{
-      complete: BooleanFilter;
-      id: NumericFilter;
-      body: StringFilter;
-    }>();
+    expectTypeOf<ObjectTypeFilter<typeof MockOntology, "Todo">>().toMatchTypeOf<
+      {
+        complete: BooleanFilter;
+        body: StringFilter;
+        id: StringFilter;
+      }
+    >();
+    expectTypeOf<ObjectTypeFilter<typeof MockOntology, "Task">>().toMatchTypeOf<
+      {
+        id: NumericFilter;
+      }
+    >();
```

Comment:
The compile errors for `.toMatchTypeOf` are really rough.

At first I thought that the old version was barfing because of a type error I made, but it was actually complaining because id should actually be a StringFilter given the current MockOntology. I added a second test case here to exercise the NumericFilter path.

Reply:
Yeah it's a bit difficult to debug them, I've just ended up making objects like 
`x = {} as any as expectedType` and seeing what x looks like to see what's missing 😅